File: api-tester/src/main/java/com/revenuecat/apitester/java/VirtualCurrencyInfoAPI.java
package com.revenuecat.apitester.java;

import com.revenuecat.purchases.ExperimentalPreviewRevenueCatPurchasesAPI;
import com.revenuecat.purchases.VirtualCurrencyInfo;

final class VirtualCurrencyInfoAPI {
    @ExperimentalPreviewRevenueCatPurchasesAPI
    static void check(final VirtualCurrencyInfo virtualCurrencyInfo) {
        final int balance = virtualCurrencyInfo.getBalance();
        final Integer balance2 = virtualCurrencyInfo.getBalance();
    }
}

File: api-tester/src/main/java/com/revenuecat/apitester/kotlin/VirtualCurrencyInfoAPI.kt
package com.revenuecat.apitester.kotlin

import com.revenuecat.purchases.ExperimentalPreviewRevenueCatPurchasesAPI
import com.revenuecat.purchases.VirtualCurrency

@Suppress("unused", "UNUSED_VARIABLE")
private class VirtualCurrencyInfoAPI {
    @OptIn(ExperimentalPreviewRevenueCatPurchasesAPI::class)
    fun check(virtualCurrencyInfo: VirtualCurrency) {
        with(virtualCurrencyInfo) {
            val balance: Int = virtualCurrencyInfo.balance
        }
    }
}
